File: anwendungslogik/src/main/java/SpookyScarySkeletons/anwendungslogik/model/Scenario.java
package SpookyScarySkeletons.anwendungslogik.model;

public class Scenario {

    private String name;
    private String websocketEndpoint;
    private String previewImageUrl;
    private String backgroundImageUrl;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getWebsocketEndpoint() {
        return websocketEndpoint;
    }

    public void setWebsocketEndpoint(String websocketEndpoint) {
        this.websocketEndpoint = websocketEndpoint;
    }

    public String getPreviewImageUrl() {return previewImageUrl; }

    public void setPreviewImageUrl(String previewImageUrl) { this.previewImageUrl = previewImageUrl; }

    public String getBackgroundImageUrl() { return backgroundImageUrl; }

    public void setBackgroundImageUrl(String backgroundImageUrl) {
        this.backgroundImageUrl = backgroundImageUrl;
    }
}

File: app/lib/model/Scenario.dart
class Scenario {
  final String name;
  String websocketEndpoint;
  final String previewImageUrl;
  final String backgroundImageUrl;

  Scenario(this.name, this.websocketEndpoint, this.previewImageUrl, this.backgroundImageUrl);

  Scenario.fromJSON(Map<String, dynamic> json)
      : name = json['name'],
        websocketEndpoint = json['websocketEndpoint'],
        previewImageUrl = json['previewImageUrl'],
        backgroundImageUrl = json['backgroundImageUrl'];

  Map<String, dynamic> toJson() => {
        'name': name,
        'websocketEndpoint': websocketEndpoint,
        'previewImageUrl': previewImageUrl,
        'backgroundImageUrl': backgroundImageUrl
      };
}
